fvtest/compilertriltest/LongAndAsRotateTest.cpp

```diff
@@ -0,0 +1,494 @@
+         "                  (@id \"myNode\")                   "
+         "                  (iconst 0x1234)                    "
+         "               )                                     "
+         "            )                                        "
+         "            (lconst %s)                              "
+         "         )                                           "
+         "      )                                              "
+         "   )                                                 "
+         ")                                                    ",  
+  */       //above tree is compacted below:
+         "(method name=\"AddAndLand\" return=Int64 args=[Int32]  "
+         "(block(lstore temp=\"x\"(land(i2l                      "
+         "(iand(iload parm=0 id=\"myNode\")(iconst 0x4321)))     "
+         "(lconst 0xFFFFFFFFFFFF0000)))                          "
+         "(lreturn(land(i2l(iand(@id \"myNode\")(iconst 0x1234)))"
+         "(lconst %s)))))                                        ",  
+         std::to_string(param.constantValue).c_str()
+         );
+
+   auto trees = parseString(inputTrees);
+
+   ASSERT_NOTNULL(trees);
+
+   Tril::JitBuilderCompiler compiler {trees};
+
+   ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+   auto entry_point = compiler.getEntryPoint<uint64_t (*)(int32_t)>();
+
+   ASSERT_EQ(param.oracle((param.variableValue & 0x1234), param.constantValue), entry_point(param.variableValue)); 
+
+   }
+     
+//This test will create the following scenario given an iload 0xFF and lconst 0x8000000000000001:
+// i2l node with highWordZero=true
+
+TEST_P(i2lLongAndAsRotateTest, iConstTest)
+   {
+   auto param = to_struct(GetParam());
+
+   char inputTrees[512] = {0};
+   std::snprintf(inputTrees, sizeof(inputTrees),
+         "(method return=Int64 args=[Int32]  "
+         " (block                            "
+         "  (lreturn                         "
+         "  (land                            " 
+         "      (i2l                         " //basic block extension will realize that this i2l will have highWordZero
+         "         (iand                     "
+         "            (iload parm=0)         "
+         "            (iconst 0x4321)        "
+         "         )                         "
+         "      )                            " 
+         "      (lconst %s)))))              ",
+         std::to_string(param.constantValue).c_str()
+         );
+
+   auto trees = parseString(inputTrees);
+
+   ASSERT_NOTNULL(trees);
+
+   Tril::JitBuilderCompiler compiler {trees};
+
+   ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+   auto entry_point = compiler.getEntryPoint<uint64_t (*)(int32_t)>();
+
+   ASSERT_EQ(param.oracle((param.variableValue & 0x4321), param.constantValue), entry_point(param.variableValue)); 
```

Comment:
Where's the 0x4321 coming from?

Reply:
I should have defined this more clearly. The 0x4321 is just a positive value that I use for the `iconst` node to get the optimizer to set the `i2l` node's highWordZero flag. 
The comment `//basic block extension will realize that this i2l will have highWordZero` was entered in relation to this. I'll make this more clear.

Reply:
Would it be useful do you think to be able to directly set the 'highWordZero' flag on a node? If so, maybe open an enhancement issue. It's something milling about in my mental backlog, but I don't have an explicit issue for it yet

Reply:
I opened an issue: https://github.com/eclipse/omr/issues/1947